Makefile

```diff
@@ -1,19 +1,43 @@
 # holochain Makefile
 
+F=RUSTFLAGS="-Dwarnings"
+
 # mark everything as phony because it doesn't represent a file-system output
-.PHONY: default build-workspace test-workspace
+.PHONY: default \
+	static-all static-fmt static-toml static-clippy static-doc \
+	build-workspace test-workspace
 
 # default to running everything (first rule)
 default: build-workspace test-workspace
 
+# execute all static code validation
+static-all: static-fmt static-toml static-clippy static-doc
+
+# ensure committed code is formatted properly
+static-fmt:
+	$(F) cargo fmt --check
+
+# lint our toml files
+static-toml:
+	cargo install taplo-cli
+	$(F) taplo format --check ./*.toml
+	$(F) taplo format --check ./crates/**/*.toml
+
+# ensure our chosen style lints are followed
+static-clippy:
+	$(F) cargo clippy
+
+# ensure we can build the docs
+static-doc:
+	$(F) cargo doc
```

Comment:
@jost-s - can you double check this? I'm not great at reading nix code, but it seemed to me this is all it was doing...

Reply:
Yes, that's all that it's doing https://crane.dev/API.html#cranelibcargodoc

Reply:
It turns out that for `cargo doc` `RUSTFLAGS` is not effective. It needs [`RUSTDOCFLAGS`](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-reads). But we can only enable it after fixing all the cross-link doc comments.